### ISSUE TITLE
0.14.0.2 not syncing

### ISSUE THREAD

**Issue by rebroad** [dashpay/dash#3040]
I have just installed a fresh 0.14.0.2, and after leaving it running for 30 minutes, it still has not downloaded any blocks. Compared to bitcoin, it would have certainly started downloading blocks by now.

**Comment by strophy** [comment]
This sounds like problems with your connection or firewall. Verify port 9999 is open and check you are not connected to any pre-0.14.0.x nodes in the peers list under Tools > Peers list. If necessary, right click the older peers and manually ban them. Once you have a few 0.14 peers sync should start very quickly.

**Comment by tgflynn** [comment]
I appear to be having the same problem after upgrading from a 0.13 version to 0.14.0.2.  I don't believe it's a network problem because I have peers and network traffic.  All peers appear to be 0.14+.  dash-qt shows status as "Connecting to peers" and number of blocks/progress does not increase.  

The standard debug log contains mostly messages like this:

`
2019-08-15 12:20:13 CMasternodeSync::ProcessTick -- nTick 162 nCurrentAsset 0 nTriedPeerCount 0 nSyncProgress -0.250000
2019-08-15 12:20:19 CMasternodeSync::ProcessTick -- nTick 168 nCurrentAsset 0 nTriedPeerCount 0 nSyncProgress -0.250000
2019-08-15 12:20:25 CMasternodeSync::ProcessTick -- nTick 174 nCurrentAsset 0 nTriedPeerCount 0 nSyncProgress -0.250000
2019-08-15 12:20:31 CMasternodeSync::ProcessTick -- nTick 180 nCurrentAsset 0 nTriedPeerCount 0 nSyncProgress -0.250000
2019-08-15 12:20:37 CMasternodeSync::ProcessTick -- nTick 186 nCurrentAsset 0 nTriedPeerCount 0 nSyncProgress -0.250000
2019-08-15 12:20:43 CMasternodeSync::ProcessTick -- nTick 192 nCurrentAsset 0 nTriedPeerCount 0 nSyncProgress -0.250000
2019-08-15 12:20:49 CMasternodeSync::ProcessTick -- nTick 198 nCurrentAsset 0 nTriedPeerCount 0 nSyncProgress -0.250000
2019-08-15 12:20:55 CMasternodeSync::ProcessTick -- nTick 204 nCurrentAsset 0 nTriedPeerCount 0 nSyncProgress -0.250000
2019-08-15 12:21:01 CMasternodeSync::ProcessTick -- nTick 210 nCurrentAsset 0 nTriedPeerCount 0 nSyncProgress -0.250000
2019-08-15 12:21:07 CMasternodeSync::ProcessTick -- nTick 216 nCurrentAsset 0 nTriedPeerCount 0 nSyncProgress -0.250000

`
I've tried deleting all cache files with no change in behavior.

I'm trying reindexing now but it looks like it's going to be very slow.  It looks like it's trying to sync while reindexing and the syncing behavior seems to be the same as before.

**Comment by nmarley** [comment]
I had the same issue 3 days ago but while syncing from scratch (no blockchain or anything, on a fresh VPS). For some reason no blocks were coming in, but I kept seeing the `CMasternodeSync::ProcessTick` log messages. Eventually blocks started coming in (after ~10-15 minutes) but not sure if it was some concurrency/thread issue or what. This was also on version 0.14.0.2.

**Comment by UdjinM6** [comment]
These are two different issues I believe:

1. Fresh sync. Your node is probably still downloading headers which is not shown in debug.log in any way unfortunately but you can check the progress via `getpeerinfo`. Could be that the peer you are connected to is too slow.

2. v13->v14 upgrade. See https://github.com/dashpay/dash/issues/2995

**Comment by tgflynn** [comment]
So the reindex run ended with a segfault (unfortunately, I didn't have core dumps enabled).  I started again without the -reindex option and eventually the syncing came unstuck and I was able to sync completely.

Note that overall though my node spent many hours in the stuck state so this isn't really a transient issue that always resolves itself in a reasonable period of time.

I'm not really convinced this is the same as #2995, at least if the explanation of network segmentation was the correct explanation for that.  For one thing a good deal of time has passed since that was reported.  Also all my peers were 0.14+ so unless there is a segment of zombie 0.14 nodes on the network I don't see how all my peers could have been providing bad data.
